app/helpers/shared_helper/ab_list_helper.rb

```diff
@@ -0,0 +1,141 @@
+        row_data(_('Message'), resolve[:new][:object_message]),
+        row_data(_('Request'), resolve[:new][:object_request])
+      ]
+    }
+    miq_structured_list(data)
+  end
+
+  def object_attribute(custom_button, resolve)
+    custom_button.enablement_expression
+    data = {
+      :title => _('Object Attribute'),
+      :mode  => _('object_attribute'),
+      :rows  => [
+        row_data(_('Type'), resolve[:new][:target_class]),
+      ]
+    }
+    if resolve[:new][:target_class].present?
+      data[:rows] << row_data(_('Selection'), resolve[:new][:target_id])
+    end
+    data[:rows] << if custom_button && !custom_button.enablement_expression.nil? && custom_button.enablement_expression.kind_of?(MiqExpression)
+                     row_data(
+                       _('Enablement expression'),
+                       custom_button_info_text(custom_button, :enablement_expression, custom_button.enablement_expression.to_human)
+                     )
+                   else
+                     row_data(
+                       _('Enablement expression'),
+                       custom_button_info_text(custom_button, :enablement_expression, _('No Enablement Expression defined, this button will be always enabled'))
+                     )
+                   end
+    data[:rows] << row_data(_('Disabled Button Text'), custom_button.disabled_text)
+    miq_structured_list(data)
+  end
+
+  def visibility_expression(custom_button)
+    data = {
+      :title => _('Visibility Expression'),
+      :mode  => _('visibility_expression'),
+      :rows  => [
+        row_data(
+          _('Visibility Expression'),
+          custom_button_info_text(custom_button, :visibility_expression, _('No Visibility Expression defined, this button will be always visible'))
+        ),
+      ]
+    }
+    miq_structured_list(data)
+  end
+
+  def custom_button_info_text(custom_button, attribute, default_message)
+    value = custom_button.send(attribute)
+    value.nil? ? _(default_message) : value
+  end
+
+  def attribute_value_pair(resolve)
+    data = {
+      :title => _('Attribute/Value Pairs'),
+      :mode  => _('attribute_value_pair'),
+      :rows  => []
+    }
+    if resolve[:new][:attrs].empty?
+      data[:rows] << row_data(_('Attribute/Value Pairs'), _('No Attribute/Value Pairs found.'))
+    else
+      resolve[:new][:attrs].each_with_index do |attr, i|
+        data[:rows] << row_data(_(i + 1), [attr[0], attr[1]])
+      end
+    end
+    miq_structured_list(data)
+  end
+
+  def visibility(custom_button)
+    data = {
+      :title => _('Visibility'),
+      :mode  => _('visibility')
+    }
+
+    rows = []
+    rows << if custom_button.visibility.nil? || (custom_button.visibility && custom_button.visibility[:roles] && custom_button.visibility[:roles][0] == "_ALL_")
+              row_data(_('Show'), _("To All"))
+            else
+              row_data(_('Show'), _("By Role"))
+            end
+
+    data[:rows] = rows
```

Comment:
can you see if this works - 
`data[:rows] = row_data(_('Show'), visibility_roles&.empty? || visibility_roles&.first == "_ALL_" ? _("To All") : _("By Role"))`

Reply:
Seems like there having a trouble to get `visibility_roles`. Alternatively, as suggested, we can use 
` row_data(_('Show'), custom_button&.visibility.empty? || (custom_button&.visibility&.fetch(:roles, [])&.first == "_ALL_") ? _("To All") : _("By Role"))`